### ISSUE TITLE
ContentLength missing in FileParameter

### ISSUE THREAD

**Issue by glecchi** [restsharp/RestSharp#814]
The AddFile method with Writer Delegate param on RestRequest doesn't set the file ContentLength. 
When Delegate write on Request stream throw a ProtocollViolationException (The number of bytes to be written to the stream exceeded Content-Length size specified) because file ContentLength  is zero.

**Comment by kfstorm** [comment]
I met the same issue. Any one have update?

**Comment by alexeyzimarev** [comment]
I guess since the method now requires to specify the length, this issue can be closed

```csharp
public IRestRequest AddFile(string name, Action<Stream> writer, string fileName, long contentLength,  string contentType = null)
```

**Comment by alexeyzimarev** [comment]
No, I am wrong, actually, you meant using `AddFile(FileParameter p)`
